src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3043,8 +3063,12 @@ void EmitSingleCharLazy(RegexNode node, RegexNode? subsequent = null, bool emitL
                 {
                     if (iterationCount is null &&
                         node.Kind is RegexNodeKind.Notonelazy &&
-                        subsequent?.FindStartingLiteral(4) is ValueTuple<char, string?, string?, bool> literal && // 5 == max optimized by IndexOfAny, and we need to reserve 1 for node.Ch
-                        !literal.Item4) // not negated; can't search for both the node.Ch and a negated subsequent char with an IndexOf* method
+                        subsequent?.FindStartingLiteral(4) is RegexNode.StartingLiteralData literal && // 5 == max optimized by IndexOfAny, and we need to reserve 1 for node.Ch
+                        !literal.Negated && // not negated; can't search for both the node.Ch and a negated subsequent char with an IndexOf* method
+                        (literal.String is not null ||
+                         literal.SetChars is not null ||
+                         literal.Range.LowInclusive == literal.Range.HighInclusive ||
```

Comment:
They're reachable.  We hit this in the case where the literal comes from a RegexNode.One, in which case FindStartingLiteral creates a StartingLiteralData with a range: (node.Ch, node.Ch) rather than allocating a new char[1] { node.Ch }.